### DISCUSSION TITLE
Invalid Credentials | or test dosn't show any popup

### DISCUSSION THREAD

**Discussion by nicedevil007** [linuxserver/Heimdall#1003] [App issues]
According to this already closed issue: https://github.com/linuxserver/Heimdall/issues/665
Is there a fix?

I have the recent version installed of heimdall and portainer and other things like nextcloud. On portainer I get "invalid credentials" now and they were working before and they are 100000% correct.

Nextcloud seems to be a bit different, no message box that the credentials are wrong, just nothing happens on the test.

**Comment by Brandoskey** [comment]
I get this too. Both heimdall and portainer are installed on the same vm. Restarting the vm allows it to work for a short time then it stops. Another instance of portainer on a separate machine has no issues

**Comment by CarbonV** [comment]
I'm having the same issue, someone said it was related to portainer rate limiting requests using login. However token access is now an option so maybe Heimdall should switch to that for the enhanced part of portainer?

**Comment by pops199** [comment]
I also thought it was the rate limiter but it looks like its the way Heimdall is written.

> NO NO NO the issue isn't that portainer is doing rate limiting. The issue is that the way Heimdall is written is not right. It should hold the session open rather than login over and over and over and over and over again.
> 
> For example I have the exact same issue when Heimdall logs into OpenMediaVault it sends an email. The issue within 10 minutes it sent out almost a THOUSAND emails. And you can't stop OpenMediaVault from emailing every time you log in. It almost went over the maximum amount of emails Mailgun can send on the free tier and in UNDER 10 minutes!!!
> 
> The issue is that Heimdall should log in ONCE and then retain that session, constantly logging in is sloppy coding, but I get why, I'm a coder too ... remembering state and AJAX requests is hard.

**Comment by eenturk** [comment]
Same problem here

**Comment by KodeStar** [comment]
It's not actually a problem with Heimdall itself, both those enhanced apps were written by 3rd parties as I don't use either app. I agree though that it sounds like they could be improved.
